pkg/hooks/manager.go

```diff
@@ -56,6 +56,19 @@ type callBackClient struct {
 var manager Manager
 var once sync.Once
 
+var DeprecatedVersions = map[string]bool{
+	hooksV1alpha3.Version: hooksV1alpha3.Deprecated,
+	hooksV1alpha2.Version: hooksV1alpha2.Deprecated,
+	hooksV1alpha1.Version: hooksV1alpha1.Deprecated,
+}
+
+// The order matters. Newest versions first.
+var SupportedVersions = []string{
+	hooksV1alpha3.Version,
+	hooksV1alpha2.Version,
+	hooksV1alpha1.Version,
+}
```

Comment:
It will be nicer to avoid changing these slices in the future here.
The versions could be registered by the individual package/file through an `init()`, leaving here only the infrastructure.

Reply:
Sorry, I did not understand your suggestion. While we are alpha/beta, what is the problem with the slice of versions and map of deprecation? What is the change you would like to see here?

Reply:
@EdDev ping on above question.

Reply:
Sorry, seems I have missed the answer.

> One problem would be the order. We can't guarantee the order using init(), can we?

The order of the `init()` calls cannot be controlled, but we can make sure items are added to the slice in an ordered manner (version names are sort-able).

I'm not that clear why we need them ordered in the first place.
Is it for picking the default/latest one when answering the client? (if that is the case, we can mark which is the "default" when registering).

Reply:
> The order of the init() calls cannot be controlled, but we can make sure items are added to the slice in an ordered manner (version names are sort-able).

Okay, will do that.

> I'm not that clear why we need them ordered in the first place.
> Is it for picking the default/latest one when answering the client? 

Yes. We receive a [list](https://github.com/kubevirt/kubevirt/blob/main/pkg/hooks/info/api_info.pb.go#L56) of implemented versions and our heuristic is to get latest one. So, some order is needed

Reply:
> It will be nicer to avoid changing these slices in the future here. The versions could be registered by the individual package/file through an `init()`, leaving here only the infrastructure.

I actually worked on this Today. We can't do it how you proposed, by implementing `init()` that call manager's `Register` function would cause a dependency cycle between each version and hook package.

As we dropped adding the warning on webhook admitter, I'm not exporting both variables anymore. I'll add this to https://github.com/kubevirt/kubevirt/issues/11215 as well.